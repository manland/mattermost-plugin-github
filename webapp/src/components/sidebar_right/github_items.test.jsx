// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {mount} from 'enzyme';

import GithubItems from './github_items';
import { RHSStates } from '../../constants';

describe('GithubItems', () => {
    const baseProps = {
        items: [],
        theme: {centerChannelColor: '#fff'},
        rhsState: RHSStates.PRS,
    };

    test('should match snapshot', () => {
        const wrapper = mount(<GithubItems {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with items', () => {
        const items = [{
            id: 1, 
            repository: {full_name: 'test'}, 
            user:{login: 'manland'}, 
            title: 'make it work', 
            html_url: 'http://mattermost.com',
            labels: [],
            reason: 'mention'
        }]
        const wrapper = mount(<GithubItems {...baseProps} items={items}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with items deleted', () => {
        const items = [{
            id: 1, 
            repository: {full_name: 'test'}, 
            user:{login: 'manland'}, 
            title: 'make it work', 
            html_url: 'http://mattermost.com',
            labels: [],
            reason: 'mention'
        },{
            id: 2, 
            repository: {full_name: 'test'}, 
            user:{login: 'manland'}, 
            title: 'make it work', 
            html_url: 'http://mattermost.com',
            labels: [],
            reason: 'mention'
        }]
        const wrapper = mount(<GithubItems {...baseProps} items={items}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('div').first().instance().style.opacity).toBe('');
        expect(wrapper.find('div').last().instance().style.opacity).toBe('');
        wrapper.setProps({...baseProps, items: [items[1]]}); 
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('div').first().instance().style.opacity).toBe('0.5');
        expect(wrapper.find('div').last().instance().style.opacity).toBe('');
    });

    test('should not delete items deleted if user refresh', () => {
        const items = [{
            id: 1, 
            repository: {full_name: 'test'}, 
            user:{login: 'manland'}, 
            title: 'make it work', 
            html_url: 'http://mattermost.com',
            labels: [],
            reason: 'mention'
        }]
        const wrapper = mount(<GithubItems {...baseProps} items={items}/>);
        expect(wrapper.find('div').first().instance().style.opacity).toBe('');
        wrapper.setProps({...baseProps});
        expect(wrapper.find('div').first().instance().style.opacity).toBe('0.5');
        wrapper.setProps({...baseProps, items: []});
        expect(wrapper.find('div').first().instance().style.opacity).toBe('0.5');
    });
});
