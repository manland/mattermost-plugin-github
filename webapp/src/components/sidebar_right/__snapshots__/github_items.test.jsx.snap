// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubItems should match snapshot 1`] = `
<GithubItems
  items={Array []}
  rhsState="pullRequests"
  theme={
    Object {
      "centerChannelColor": "#fff",
    }
  }
>
  <div
    style={
      Object {
        "borderTop": "1px solid rgba(255,255,255,0.2)",
        "padding": "15px",
      }
    }
  >
    You have no active items
  </div>
</GithubItems>
`;

exports[`GithubItems should match snapshot with items 1`] = `
<GithubItems
  items={
    Array [
      Object {
        "html_url": "http://mattermost.com",
        "id": 1,
        "labels": Array [],
        "reason": "mention",
        "repository": Object {
          "full_name": "test",
        },
        "title": "make it work",
        "user": Object {
          "login": "manland",
        },
      },
    ]
  }
  rhsState="pullRequests"
  theme={
    Object {
      "centerChannelColor": "#fff",
    }
  }
>
  <div
    key="1"
    style={
      Object {
        "borderTop": "1px solid rgba(255,255,255,0.2)",
        "padding": "15px",
      }
    }
  >
    <div>
      <strong>
        <a
          href="http://mattermost.com"
          rel="noopener noreferrer"
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "lineHeight": 1.7,
            }
          }
          target="_blank"
        >
          make it work
        </a>
      </strong>
    </div>
    <GithubLabels
      labels={Array []}
    />
    <div
      className="light"
      style={
        Object {
          "fontSize": "13px",
          "margin": "5px 0 0 0",
        }
      }
    >
      Created by manland 
      at test.
      <br />
      You were specifically @mentioned in the content.
    </div>
  </div>
</GithubItems>
`;

exports[`GithubItems should match snapshot with items deleted 1`] = `
<GithubItems
  items={
    Array [
      Object {
        "html_url": "http://mattermost.com",
        "id": 1,
        "labels": Array [],
        "reason": "mention",
        "repository": Object {
          "full_name": "test",
        },
        "title": "make it work",
        "user": Object {
          "login": "manland",
        },
      },
      Object {
        "html_url": "http://mattermost.com",
        "id": 2,
        "labels": Array [],
        "reason": "mention",
        "repository": Object {
          "full_name": "test",
        },
        "title": "make it work",
        "user": Object {
          "login": "manland",
        },
      },
    ]
  }
  rhsState="pullRequests"
  theme={
    Object {
      "centerChannelColor": "#fff",
    }
  }
>
  <div
    key="1"
    style={
      Object {
        "borderTop": "1px solid rgba(255,255,255,0.2)",
        "padding": "15px",
      }
    }
  >
    <div>
      <strong>
        <a
          href="http://mattermost.com"
          rel="noopener noreferrer"
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "lineHeight": 1.7,
            }
          }
          target="_blank"
        >
          make it work
        </a>
      </strong>
    </div>
    <GithubLabels
      labels={Array []}
    />
    <div
      className="light"
      style={
        Object {
          "fontSize": "13px",
          "margin": "5px 0 0 0",
        }
      }
    >
      Created by manland 
      at test.
      <br />
      You were specifically @mentioned in the content.
    </div>
  </div>
  <div
    key="2"
    style={
      Object {
        "borderTop": "1px solid rgba(255,255,255,0.2)",
        "padding": "15px",
      }
    }
  >
    <div>
      <strong>
        <a
          href="http://mattermost.com"
          rel="noopener noreferrer"
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "lineHeight": 1.7,
            }
          }
          target="_blank"
        >
          make it work
        </a>
      </strong>
    </div>
    <GithubLabels
      labels={Array []}
    />
    <div
      className="light"
      style={
        Object {
          "fontSize": "13px",
          "margin": "5px 0 0 0",
        }
      }
    >
      Created by manland 
      at test.
      <br />
      You were specifically @mentioned in the content.
    </div>
  </div>
</GithubItems>
`;

exports[`GithubItems should match snapshot with items deleted 2`] = `
<GithubItems
  items={
    Array [
      Object {
        "html_url": "http://mattermost.com",
        "id": 2,
        "labels": Array [],
        "reason": "mention",
        "repository": Object {
          "full_name": "test",
        },
        "title": "make it work",
        "user": Object {
          "login": "manland",
        },
      },
    ]
  }
  rhsState="pullRequests"
  theme={
    Object {
      "centerChannelColor": "#fff",
    }
  }
>
  <div
    key="1"
    style={
      Object {
        "borderTop": "1px solid rgba(255,255,255,0.2)",
        "opacity": 0.5,
        "padding": "15px",
      }
    }
  >
    <div>
      <strong>
        <a
          href="http://mattermost.com"
          rel="noopener noreferrer"
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "lineHeight": 1.7,
            }
          }
          target="_blank"
        >
          make it work
        </a>
      </strong>
    </div>
    <GithubLabels
      labels={Array []}
    />
    <div
      className="light"
      style={
        Object {
          "fontSize": "13px",
          "margin": "5px 0 0 0",
        }
      }
    >
      Created by manland 
      at test.
      <br />
      You were specifically @mentioned in the content.
    </div>
  </div>
  <div
    key="2"
    style={
      Object {
        "borderTop": "1px solid rgba(255,255,255,0.2)",
        "padding": "15px",
      }
    }
  >
    <div>
      <strong>
        <a
          href="http://mattermost.com"
          rel="noopener noreferrer"
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "lineHeight": 1.7,
            }
          }
          target="_blank"
        >
          make it work
        </a>
      </strong>
    </div>
    <GithubLabels
      labels={Array []}
    />
    <div
      className="light"
      style={
        Object {
          "fontSize": "13px",
          "margin": "5px 0 0 0",
        }
      }
    >
      Created by manland 
      at test.
      <br />
      You were specifically @mentioned in the content.
    </div>
  </div>
</GithubItems>
`;
